CMakeLists.txt

```diff
@@ -189,13 +194,13 @@ add_dependencies(hebiros_node hebiros_generate_messages_cpp)
 #   ${catkin_LIBRARIES}
 # )
 
-target_link_libraries(hebiros_node ${catkin_LIBRARIES} ${CMAKE_SOURCE_DIR}/${PROJECT_NAME}/lib/linux_x86_64/libhebi.so)
+target_link_libraries(hebiros_node ${catkin_LIBRARIES} ${CMAKE_SOURCE_DIR}/lib/linux_x86_64/libhebi.so)
```

Comment:
Removing this ${PROJECT_NAME} variable broke the build on our two test systems here; I now get the error:

`No rule to make target /path/to/workspace/src/lib/linux_x86_64/libhebi.so', needed by '/path/to/workspace/devel/lib/hebiros/hebiros_node`

(before the "PROJECT NAME" variable added in the `hebiros` subdirectory here)

I'd like to understand a bit more about why this was required in your setup?

Reply:
Hello,
If you use ${PROJECT_SOURCE_DIR} instead of ${CMAKE_SOURCE_DIR} in CMakeList.txt, you will be able to solve the project name issue. But it is not necessary to do it and I recommend to move all the files in hebiros package (in the HEBI-ROS dir!) into hebiros subdirectory to prevent changing the cmake dir name which is currently directory affected by the name of user's working dir, and also move your moveit package out side of the hebiros package because a package in another package has not been correctly recognized by the ROS file system. Please refer to our development-nbc-5dof branch how we managing this issue.
https://github.com/NihonBinary/HEBI-ROS/tree/develop-nbc-5dof
I hope this info will help you.